AdminNotificationManagerPlugin.inc.php

```diff
@@ -146,19 +148,33 @@ function getTemplatePath($inCore = false) {
 	 * @return array
 	 */
 	function _getAdminList() {
-		$user = null;
+		$arrayOfContexts = $this->_getContexts();
 		$users = array();
 
-		$roleDAO = DAORegistry::getDAO('RoleDAO');
-		$userDAO = $roleDAO->getUsersByRoleId(ROLE_ID_SITE_ADMIN);
+		foreach($arrayOfContexts as $context){
+		$user = null;
+		$userId = array();
+		$userGroupDAO = DAORegistry::getDAO('UserGroupDAO');
+		$userFactory = new PKPUserService();
+  
+		$userGroupDAOFactory = $userGroupDAO->getByRoleId($context,ROLE_ID_SITE_ADMIN);
 
-		if ($userDAO && $userDAO->getCount() > 0) {
-			while ($user = $userDAO->next()) {
-				$users[$user->getId()] = $user;
+		if ($userGroupDAOFactory && $userGroupDAOFactory->getCount() >= 1){
+			while($group = $userGroupDAOFactory->next()){
+				$groupId = $group->getId();
+				$userId[] = $groupId;     
 			}
 		}
-		return $users;
+
+			$args = array('userGroupIds'=>$userId);
+			$listOfUsers= $userFactory->getMany($args);
+			foreach($listOfUsers as $user){
+				$idValue = $user->getId();
```

Comment:
`$idValue` is only used in two lines.  Can it be simplified?  This may be valuable if it helps for readability, but may also just be cruft.

Reply:
I felt the `$idValue` was valuable for readability in this case. The name helps reduce confusion in the cases it's used, as we see exactly what value is being stored within the variable, and later when we see this being stored in an array. Overall I think it's more valuable to include than to not